src/sherpa_ai/memory/belief.py

```diff
@@ -90,6 +90,36 @@ def get_internal_history(
 
         return context
 
+    def get_histories_excluding_types(
+        self, token_counter: Callable[[str], int],  exclude_type: [EventType] , max_tokens=4000 
```

Comment:
if `exlude_type` is meant to be a single variable then annotate with `exclude_type:  EventType`. If it means a list, then use plural with `exclude_types: List[EventType]`